### PR TITLE
 Implement blf-balancing-torque-control

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
~⚠️ To be tested also in simulation~

### CODE REVIEW

**Comment by isorrentino** [comment]
Here is a video of the test in simulation

https://github.com/ami-iit/bipedal-locomotion-framework/assets/43743081/f7920f36-6196-4b9f-af76-f50d9ed8cd32

File: utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini [hunk 1]
```diff
@@ -0,0 +1,22 @@
+dt                       0.01   # (0.01 seconds)
+contact_force_threshold  0.1    # in Newton
+
+
+com_knots_delta_x        (0.0, 0.0,  0.03,  0.03, -0.03, -0.03, 0.0,  0.0)    # in meter
+com_knots_delta_y        (0.0, 0.05, 0.05, -0.05, -0.05,  0.05, 0.05, 0.0)    # in meter
+com_knots_delta_z        (0.0, 0.0,  0.0,   0.0,   0.0,   0.0,  0.0,  0.0)    # in meter
+#com_knots_delta_x        (0.0, 0.0,  0.0,  0.0, -0.0, -0.0, 0.0,  0.0)    # in meter
```
Comment (isorrentino) [inline utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini:8]:
```suggestion
```

File: utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#com_knots_delta_y        (0.0, 0.0, 0.0, -0.0, -0.0,  0.0, 0.0, 0.0)    # in meter
```
Comment (isorrentino) [inline utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini:9]:
```suggestion
```

File: utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#com_knots_delta_z        (0.0, 0.0,  0.0,   0.0,   0.0,   0.0,  0.0,  0.0)    # in meter
```
Comment (isorrentino) [inline utilities/balancing-torque-control/config/robots/ergoCubGazeboV1/blf-balancing-torque-control-options.ini:10]:
```suggestion
```

File: utilities/balancing-torque-control/script/blf-balancing-torque-control.py [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+
+
+def main():
+    # Before everything let use the YarpSink for the logger and the YarpClock as clock. These are functionalities
+    # exposed by blf.
+    # if not blf.text_logging.LoggerBuilder.set_factory(
+    #     blf.text_logging.YarpLoggerFactory("balancing-torque-control")
+    # ):
+    #     blf.log().error("Impossible to set the logger factory.")
+    #     return False
```
Comment (isorrentino) [inline utilities/balancing-torque-control/script/blf-balancing-torque-control.py:484]: I do not know if you want to re-enable the logging. If not, you may cancel out these lines.

File: utilities/balancing-torque-control/script/blf-balancing-torque-control.py [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+    if not blf.system.ClockBuilder.set_factory(blf.system.YarpClockFactory()):
+        blf.log().error("Impossible to set the logger factory.")
+        return False
+
+
```
Comment (isorrentino) [inline utilities/balancing-torque-control/script/blf-balancing-torque-control.py:489]:
```suggestion
```

**Review by isorrentino — CHANGES_REQUESTED** [review]

File: utilities/balancing-torque-control/script/blf-balancing-torque-control.py (hunk 4 above)
Comment (GiulioRomualdi) [reply utilities/balancing-torque-control/script/blf-balancing-torque-control.py:484]: I would say yes!

**Review by isorrentino — APPROVED** [review]

**Pull request merged**